pkg/validation/spec/gnostic.go

```diff
@@ -0,0 +1,1226 @@
+package spec
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/go-openapi/jsonreference"
+	openapi_v2 "github.com/googleapis/gnostic/openapiv2"
+)
+
+// Interfaces
+type GnosticCommonValidations interface {
+	GetMaximum() float64
+	GetExclusiveMaximum() bool
+	GetMinimum() float64
+	GetExclusiveMinimum() bool
+	GetMaxLength() int64
+	GetMinLength() int64
+	GetPattern() string
+	GetMaxItems() int64
+	GetMinItems() int64
+	GetUniqueItems() bool
+	GetMultipleOf() float64
+}
+
+type GnosticCommonValidationsNamedEnum interface {
+	GetEnum() []*openapi_v2.NamedAny
+}
+
+type GnosticCommonValidationsAnyEnum interface {
+	GetEnum() []*openapi_v2.Any
+}
+
+
+func (k *CommonValidations) FromGnostic(g GnosticCommonValidations) error {
+	if g == nil {
+		return nil
+	}
+
+	max := g.GetMaximum()
+	if max != 0 {
+		k.Maximum = &max
```

Comment:
does this mean `maximum: 0` does not round-trip?

Reply:
That is correct. Maximum: 0 does not round trip. I am not sure how to fix this.

Reply:
Yeah, that's not new for clients that download protobuf though. Also, I was talking to @timburks about that, we can probably fix it in openapi v3.0.

Reply:
@apelisse can you elaborate on  “that’s not for new clients”

Reply:
Clients that use the protobuf version (basically kubectl AFAIK), already can't distinguish between 0 and unset. So that's not a new behavior. This is only going to affect clients that start doing the conversion from gnostic to swagger, and there are none today. And it's not something we use today in `kubectl explain` for example (we don't take advantage of this information in there, yet).

Reply:
ok